### PR TITLE
Recovering deleted files using Foremost

### LINKED ISSUE #3183: Recovering Deleted Files Using Foremost
### Proposed title of article

Recover Deleted Files With Foremost

### Introduction paragraph (2-3 paragraphs):
Foremost is a forensic program used to recover lost files based on their headers, footers, and internal data structures. It implements a technique known as data carving. It is free and comes with most Linux platforms. This makes it ideal to use in recovering deleted data in hard disk drives and pen drives.

In this tutorial, we will be showing stepwise how to use foremost to recover deleted files from a pen drive. We will then show how to use foremost to recover deleted files from a whole hard disk drive or a specific partition of the hard drive.

### Key takeaways: 

1. Installing foremost in a Linux platform.
2. Tools included in the foremost package.
3. Use foremost to recover deleted files of all types.
4. Use foremost to recover a specific file type.

### References: 

N/A

### PR DESCRIPTION
**Pull request by FINCH285**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3183 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by FINCH285** [comment]
@Bayler @hectorkambow I've worked on the plagiarism issues, you can continue with the review

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/recover-deleted-files-with-foremost/index.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+### Prerequisites
+
+To understand this article a reader needs to have:
+
+- Kali Linux or any other Linux distribution installed.
+- Prior knowledge of the basic commands in the Linux terminal.
+
+Check the following article to understand the prerequisites.
+[Getting started with kali linux](https://www.section.io/engineering-education/getting-started-with-kali-linux/)
```
Comment (Bayler) [inline content/articles/recover-deleted-files-with-foremost/index.md:19]: Please note that the format for linking a Section resource is as follows:

[Natural Language Processing](/engineering-education/five-real-life-use-cases-of-natural-language-processing-nlp/)

File: content/articles/recover-deleted-files-with-foremost/index.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+### Installing foremost in Kali Linux
+
+Foremost comes pre-installed in most Linux distributions. If it is not installed, type the following command in the terminal. After executing the command **foremost** will be installed.
+
```
Comment (Bayler) [inline content/articles/recover-deleted-files-with-foremost/index.md:24]: Please let the reader know how to check if foremost is pre-installed in their machine. So, that if it's not installed, they type in the command

File: content/articles/recover-deleted-files-with-foremost/index.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+```bash
+sudo apt install foremost
+```
+
+### Foremost package tools
+
+**Foremost** package includes some flags. We use the following command to display them.
+
```
Comment (Bayler) [inline content/articles/recover-deleted-files-with-foremost/index.md:32]: Please explain to the reader what a flag is and its purpose.

File: content/articles/recover-deleted-files-with-foremost/index.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+The above command gives the following output.
+
+![List Drives](/engineering-education/recover-deleted-files-with-foremost/drives.png)
+
+Scroll down to where your USB is located and copy the file path. To start the file recovery process, use the command below:
+
+```bash
+sudo foremost -v -t jpg,pdf,mp4,exe -q -i /dev/sdc -o /home/Desktop/recover -T
+```
```
Comment (Bayler) [inline content/articles/recover-deleted-files-with-foremost/index.md:57]: Is this file path bound to change depending on the users machine? Let it be clear to the reader

File: content/articles/recover-deleted-files-with-foremost/index.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+-t flag specifies the file types to be recovered as shown earlier. Type the file types you would like to recover. -o flag specifies the output file path. -i specifies the drive you want to recover files from. Replace the USB path with yours.
+The command gives the following output.
+
+![start recovery](/engineering-education/recover-deleted-files-with-foremost/recover.png)
+
+After the recovery process has run to completion, the recovered files will be saved in the output file path. You may need to practice some patience as the process may take longer than usual depending on the drive's size. Below is the output file.
+
+![Output](/engineering-education/recover-deleted-files-with-foremost/output.png)
+
```
Comment (Bayler) [inline content/articles/recover-deleted-files-with-foremost/index.md:67]: Please interpret for the reader what this results mean

File: content/articles/recover-deleted-files-with-foremost/index.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+**To recover files from the entire hard drive use the following command:**
+
+```bash
+sudo foremost -v -t png,mp4,pdf  -i /dev/sda -o /home/Desktop/recover -T
+```
+
+Replace the hard drive path with your own. You will find the path in the output of the command used to list drives. Below is an output of the command above.
+
+![Start recovery](/engineering-education/recover-deleted-files-with-foremost/disk.png)
+
```
Comment (Bayler) [inline content/articles/recover-deleted-files-with-foremost/index.md:94]: Are these the recovered files? Please interpret these results to the reader

File: content/articles/recover-deleted-files-with-foremost/index.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+```bash
+sudo foremost -v -t png,mp4,pdf -i /dev/sda1  -o /home/Desktop/recover -T
+```
+
+Replace the partition path with your own. This process will take a while depending on the size of the partition. For larger partitions, be patient as this process may take some time.
+
+### Recovering files of all types
+
+To recover all the file types that foremost supports, use the following command.
```
Comment (Bayler) [inline content/articles/recover-deleted-files-with-foremost/index.md:107]: Are we recovering all the files or the file types?

File: content/articles/recover-deleted-files-with-foremost/index.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+Unintentional deletion of important files can be tragic. This has led to operating systems having a restore point. In Linux, the restore point is the trash. In Windows, the restore point is the recycle bin. Sometimes we delete these files in the restore points too. At this point, we opt to visit data recovery centers or buy expensive data recovery software.
+
+### Introduction
+
+[**Foremost**](https://pkgs.org/download/foremost) is a free data retrieval software. You can download it on the internet. It operates on Linux operating systems. Throughout this tutorial, we will be using Kali Linux. Anyone using any other Linux distribution can follow through.
+
+In this tutorial, we shall be looking at the use of **foremost** in data recovery of lost/deleted files from a USB drive. We shall then go through how to recover deleted files from a whole hard disk. Finally, we will learn how to recover deleted files from a particular partition of a hard drive.
+
+### Prerequisites
+
```
Comment (Bayler) [inline content/articles/recover-deleted-files-with-foremost/index.md:12]: It is always advisable to include a Table of contents/Outline section. Please include one

**Comment by Bayler** [comment]
Hi @FINCH285 This was a nice read. Please work on the revisions then we can proceed for final review.

File: content/articles/recover-deleted-files-with-foremost/index.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+Replace the drive path above with the one you would like to recover files from.
+
+The above command will automatically generate an output folder called output. The location of the folder will be in the home directory. The output folder is shown in the picture below.
+
+![Output](/engineering-education/recover-deleted-files-with-foremost/all.png)
+
+### Conclusion
+
+We have discussed the numerous ways you can use **foremost** to recover your deleted files. There are no limitations to what you can do with **foremost**. Use the tools we discussed above to your preference.
```
Comment (Bayler) [inline content/articles/recover-deleted-files-with-foremost/index.md:121]: I'm not satisfied with your conclusion statement. Write your final point clearly, providing closure to the reader, leaving them with a strong impression of the significance of using foremost within a broader context. Build up on what you've already written.

**Comment by FINCH285** [comment]
Hi @Bayler Worked on the revisions.

File: content/articles/recover-deleted-files-with-foremost/index.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+### What we will learn
+
+In this tutorial, we shall be looking at:
+- Installation of **foremost** in Kali Linux.
+- Tools included in the **foremost** package and their use.
+- Use of **foremost** in data recovery of lost/deleted files from a USB drive.
+- Recovering deleted files from a whole hard disk. 
+- Recovering deleted files from a particular partition of a hard drive.
+- Recovering files of all types.
+
```
Comment (Bayler) [inline content/articles/recover-deleted-files-with-foremost/index.md:18]: The format for writing your `Table of contents` section is wrong. Please follow the format in this [tutorial](https://github.com/section-engineering-education/engineering-education/pull/3310/files)

**Comment by FINCH285** [comment]
@Bayler Requested change done.

**Comment by Bayler** [comment]
Hi @FINCH285 Welcome again to our Engineering program. Your tutorial is now ready for final review. Please take note of the changes I've made on my commits so that you can better your next tutorial.

Note: The Front matter and PRC line are added by the peer reviewer.

**Comment by LinusMuema** [comment]
Hi @FINCH285 
Thank you for this amazing article. I can see @Bayler has given you a warm welcome :smile:. We hope to see more great content like this from you. Awesome job!

cc: @Bayler

**Comment by FINCH285** [comment]
Hi @LinusMuema 
Thank you for the compliment. You are right @Bayler was very welcoming😃.

**Pull request merged**